command/build_docker_test.go

```diff
@@ -0,0 +1,75 @@
+package command
+
+import (
+	"os"
+	"bytes"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/packer/builder/docker"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/provisioner/file"
+	"github.com/hashicorp/packer/provisioner/shell"
+	shell_local "github.com/hashicorp/packer/provisioner/shell-local"
+)
+
+func dockerPackerMeta(*testing.T) Meta {
+	packerCfg := packer.CoreConfig{}
+
+	packerCfg.Components = packer.ComponentFinder{
+		Builder: func(n string) (packer.Builder, error) {
+			switch n {
+			case "docker":
+				return &docker.Builder{}, nil
+			default:
+				panic(n)
+			}
+		},
+		Provisioner: func(n string) (packer.Provisioner, error) {
+			switch n {
+			case "shell-local":
+				return &shell_local.Provisioner{}, nil
+			case "shell":
+				return &shell.Provisioner{}, nil
+			case "file":
+				return &file.Provisioner{}, nil
+			default:
+				panic(n)
+			}
+		},
+	}
+	var out, err bytes.Buffer
+	return Meta{
+		CoreConfig: &packerCfg,
+		Ui: &packer.BasicUi{
+			Writer:      &out,
+			ErrorWriter: &err,
+		},
+	}
+}
+
+func TestBuildDocker(t *testing.T) {
+	if v:= os.Getenv("CIRCLECI"); v == "" {
+		t.Skipf("CIRCLECI '%s'. skipping", v)
```

Comment:
@SwampDragons, what do you think of this one ? I wanted to test #6190 but only circle-ci can docker. Also this feels more like an integration test 🤔. May be I should rather put it in another test that is ran nightly or something ?

I'm not sure that by putting this here I start a something that becomes a burden to maintain, etc.